### ISSUE TITLE
RegexpError in ntriples/writer

### ISSUE THREAD

**Issue by jcoyne** [ruby-rdf/rdf#167]
```
/usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/ntriples/writer.rb:58: warning: character class has ']' without escape: /\A[ -!#-[]-~]*\z/
/usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/ntriples/writer.rb:58:in `=~': premature end of char-class: /\A[\x20-\x21\x23-\x5B\x5D-\x7E]*\z/m (RegexpError)
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/ntriples/writer.rb:58:in `escape'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/ntriples/writer.rb:280:in `escaped'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/ntriples/writer.rb:268:in `format_literal'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/writer.rb:427:in `format_term'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/ntriples/writer.rb:171:in `serialize'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/ntriples.rb:65:in `serialize'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/model/literal.rb:418:in `inspect'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/model/literal/numeric.rb:39:in `=='
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/model/list.rb:111:in `!='
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/model/list.rb:111:in `valid?'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:433:in `is_valid_list?'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:450:in `collection'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:499:in `path'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:522:in `block in objectList'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:516:in `each'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:516:in `each_with_index'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:516:in `objectList'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:543:in `block in predicateObjectList'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:539:in `each'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:539:in `each_with_index'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:539:in `predicateObjectList'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:571:in `triples'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:579:in `statement'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:166:in `block in write_epilogue'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:164:in `each'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:164:in `write_epilogue'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/writer.rb:226:in `initialize'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-turtle-1.1.5/lib/rdf/turtle/writer.rb:100:in `initialize'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/writer.rb:153:in `new'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/writer.rb:153:in `block in buffer'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/writer.rb:151:in `open'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/writer.rb:151:in `buffer'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/writer.rb:130:in `dump'
	from /usr/local/lib/ruby/gems/2.2.0/gems/rdf-1.1.7/lib/rdf/mixin/enumerable.rb:678:in `dump'
```

**Comment by jcoyne** [comment]
I checked HEAD (91ebca69cb6b) and it also has this error

**Comment by jcoyne** [comment]
I meant to say, it's still present in 1d65a2ff844a too.

**Comment by jcoyne** [comment]
@gkellogg Is this due to https://bugs.ruby-lang.org/issues/10670?

**Comment by jcoyne** [comment]
Here's what the graph looks like in rdfxml:
```ruby
resource.dump(:rdfxml)
=> "<?xml version='1.0' encoding='utf-8' ?>\n<rdf:RDF xmlns:ns0='info:fedora/fedora-system:def/model#' xmlns:ns1='http://purl.org/dc/terms/' xmlns:rdf='http://www.w3.org/1999/02/22-rdf-syntax-ns#'>\n  <rdf:Description rdf:about=''>\n    <ns1:issued rdf:datatype='http://www.w3.org/2001/XMLSchema#integer'>1922</ns1:issued>\n    <ns0:hasModel>Image</ns0:hasModel>\n  </rdf:Description>\n</rdf:RDF>\n"
```

**Comment by gkellogg** [comment]
Yes, it's a Ruby 2.2 bug. Not worth working around, it will be fixed in their next release.

**Comment by gkellogg** [comment]
Looks like it's fixed in Ruby 2.2.1.
